tests/drivers/uart/uart_async_api/src/test_uart_async.c

```diff
@@ -79,14 +79,34 @@ static void uart_async_test_init(void)
 
 struct test_data {
 	volatile uint32_t tx_aborted_count;
-	uint8_t rx_buf[5];
-	bool rx_buf_req_done;
-	bool supply_next_buffer;
-	uint8_t *last_rx_buf;
+	uint8_t rx_first_buffer[10];
+	uint32_t recv_bytes_first_buffer;
+	uint8_t rx_second_buffer[5];
+	uint32_t recv_bytes_second_buffer;
+	bool supply_second_buffer;
 };
 
 ZTEST_BMEM struct test_data tdata;
 
+/* After RX_BUF_RELEASED is signalled, the first RX_RDY will still refer to the first */
+/* buffer... Following RX_RDY calls will refer to the second buffer. Thus we need the */
+/* function below to check whether an RX_RDY event references first or second buffer. */
+/* Perhaps the async api could be changed so that RX_RDY calls after RX_BUF_RELEASED */
+/* will always refer to the next buffer. */
+
+static bool test_single_read_is_buf_in_first_buffer(const uint8_t *buf)
+{
+	uint32_t addr_buf = (uint32_t)buf;
```

Comment:
Need to use `uintptr_t` instead to support 64-bit addressing.

Reply:
thanks, I updated the PR.